apps/arm/src/main.cpp

```diff
@@ -218,8 +218,9 @@ void rxCANProcessor() {
 
   while (true) {
     if (can1.read(rxMsg)) {
-      if (canHandlerMap.count(rxMsg.id) > 0) {
-        canHandlerMap[rxMsg.id](rxMsg);
+      auto ret = canHandlerMap.at(rxMsg.id);
+      if (ret.has_value()) {
+        ret.value()(rxMsg);
```

Comment:
this could be shorter:

```suggestion
      if (auto can_function_ptr = canHandlerMap.at(rxMsg.id)) {
        can_function_ptr.value()(rxMsg);
```

Reply:
v cool c++17 :0

Reply:
pretty sure this doesn't use anything c++17 related lol. just expressions being evaluated as bools

Reply:
oh rly, i thought this was if-statement with initialization maybe im just misunderstanding what that actually is

Reply:
99% sure assignment in a conditional even works in c